leaderboard/src/lib.rs

```diff
@@ -44,18 +44,26 @@ async fn create_leaderboard(
     let raw = serde_json::to_string(&lb).map_err(|e| RpcError::Ser(e.to_string()))?;
     let key = format!("leaderboard.{}", leaderboard.id);
     let kv = KeyValueSender::new();
-    kv.set(
-        ctx,
-        &SetRequest {
-            key,
-            value: raw,
-            expires: 0,
-        },
-    )
-    .await?;
+
+    let Ok(res) = kv.get(ctx, &format!("leaderboard.{}", leaderboard.id)).await else {
+        return Ok(HttpResponse::not_found())
+    };
 
     let mut list = get_leaderboard_list(ctx).await;
-    list.push(leaderboard);
+
+    if !res.exists {
```

Comment:
aaaand that's what I get for not reading 2 lines ahead. Never mind! Leaving comment just in case others consider the same.